src/__tests__/app.integration.test.js

```diff
@@ -199,3 +226,127 @@ describe('POST /registration-requests/', () => {
+      conversationId: CONVERSATION_ID,
+      nhsNumber: NHS_NUMBER,
+      odsCode: ODS_CODE,
+      ehrRequestId: EHR_REQUEST_ID
+    });
+
+    const MODIFIED_UK06 = sendCore.mock.calls[0][2];
+
+    // then
+    expect(validateMessageEquality(ORIGINAL_UK06, MODIFIED_UK06)).toBe(true);
+    expect(response).toEqual(DEFAULT_RESULT);
+  });
+
+  it('should verify that a fragment with no external attachments is unchanged by xml changes', async () => {
+    // given
+    const NHS_NUMBER = 9693643038;
+    const CONVERSATION_ID = "05E36C93-2DEF-4586-B842-127C534FB8B7";
+    const ORIGINAL_FRAGMENTS = {
+      ['DD92589A-B5B4-4492-AADD-51534821F07B']: readFile('COPC_IN000001UK01_01', 'equality-test', 'large-ehr-no-external-attachments', 'original'),
+      ['DE4A9436-FFA3-49B1-8180-5570510F0C11']: readFile('COPC_IN000001UK01_02', 'equality-test', 'large-ehr-no-external-attachments', 'original'),
+      ['62330782-1C8B-45CD-95E3-4FC624091C61']: readFile('COPC_IN000001UK01_03', 'equality-test', 'large-ehr-no-external-attachments', 'original'),
+      ['770C42DD-301B-4177-A78E-0E9E62F3FDA1']: readFile('COPC_IN000001UK01_04', 'equality-test', 'large-ehr-no-external-attachments', 'original')
+    }
+
+    // when
+    getAllFragmentsWithMessageIdsFromRepo.mockResolvedValueOnce(Promise.resolve(ORIGINAL_FRAGMENTS));
+    sendFragment.mockResolvedValue(undefined);
+
+    await transferOutFragments({
+      conversationId: CONVERSATION_ID,
+      nhsNumber: NHS_NUMBER,
+      odsCode: ODS_CODE,
+    });
+
+    // then
+    const receivedArguments = [0, 1, 2, 3].map(i => sendFragment.mock.calls[i][2]).sort();
+    const originalFragments = Object.values(ORIGINAL_FRAGMENTS).sort();
+
+    expect(sendFragment).toBeCalledTimes(4);
+    expect(validateMessageEquality(originalFragments[0], receivedArguments[0])).toBe(true);
+    expect(validateMessageEquality(originalFragments[1], receivedArguments[1])).toBe(true);
+    expect(validateMessageEquality(originalFragments[2], receivedArguments[2])).toBe(true);
+    expect(validateMessageEquality(originalFragments[3], receivedArguments[3])).toBe(true);
+  });
+
+  it('should verify that a fragment with external attachments is unchanged by xml changes', async () => {
+    // given
+    const NHS_NUMBER = 9693643038;
+    const CONVERSATION_ID = "0346D9CC-F472-492A-86D4-43D8B73AC95D";
+    const ORIGINAL_FRAGMENTS = {
+      ['060FA820-A231-11ED-808B-AC162D1F16F0']: readFile('COPC_IN000001UK01_01', 'equality-test', 'large-ehr-with-external-attachments', 'original'),
+      ['063817B0-A231-11ED-808B-AC162D1F16F0']: readFile('COPC_IN000001UK01_02', 'equality-test', 'large-ehr-with-external-attachments', 'original'),
+      ['0635CDC0-A231-11ED-808B-AC162D1F16F0']: readFile('COPC_IN000001UK01_03', 'equality-test', 'large-ehr-with-external-attachments', 'original'),
+      ['0635CDC1-A231-11ED-808B-AC162D1F16F0']: readFile('COPC_IN000001UK01_04', 'equality-test', 'large-ehr-with-external-attachments', 'original')
```

Comment:
Lets change the keys from single element array to just a string? 
e.g. `['060FA820-A231-11ED-808B-AC162D1F16F0']`  to `'060FA820-A231-11ED-808B-AC162D1F16F0'`
In javascript it will convert all key value to a string so test would still pass, but using an array as a key isn't very good